icu4c/source/i18n/number_longnames.cpp

```diff
@@ -232,15 +282,96 @@ UnicodeString getPerUnitFormat(const Locale& locale, const UNumberUnitWidth &wid
     return UnicodeString(ptr, len);
 }
 
+// Load data originating from a deriveComponent CLDR element. The feature and
+// structure parameters must be null-terminated.
+void getDeriveComponentRule(Locale locale,
+                            const char *feature,
+                            const char *structure,
+                            UnicodeString &value0,
+                            UnicodeString &value1,
```

Comment:
I prefer putting those returned value on struct such as `struct CLDRDerivedComponent`

Thus will help us in the future to add more values.

Reply:
I guess that is necessary in Java, because Java doesn't have output parameters - in Java one must either modify an object passed as a parameter, or return an object that can encapsulate the data in question.

I looked over some ICU4C and ICU4J code to try to get a sense of what's idiomatic, but maybe @sffc is quicker in advising here: I hope with only two strings, having value0 and value1 parameters is idiomatic enough in C++ - I don't yet know what's idiomatic in ICU4J.

One random sample I found: SpoofChecker results:
https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/text/SpoofChecker.CheckResult.html
In C++:
https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/uspoof_8h.html
It's quite a bit more complicated than just two strings though, and I don't think something like that is worth the overhead for a function as simple as getDeriveComponentRule.